### PR TITLE
DPM 199 Data Hub API logging

### PR DESCRIPTION
**Pull request by marijnkampf**
### Description of change

Change logging to ASIM format.

### Checklist

* [x] Has this branch been rebased on top of the current `main` branch?

  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>

* [x] Is the CircleCI build passing?

### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/main/docs/CONTRIBUTING.md) for more guidelines.

### CODE REVIEW

File: config/settings/common_logging.py [hunk 1]
```diff
@@ -13,30 +14,30 @@
         'verbose': {
             'format': '%(asctime)s [%(levelname)s] [%(name)s] %(message)s'
         },
-        'ecs_formatter': {
-            '()': ECSFormatter,
-        },
+        "asim_formatter": {
+            "()": ASIMFormatter,
+        },
```
Comment (oliverjwroberts) [inline config/settings/common_logging.py:19]: Flake8 will probably complain about this whitespace

File: requirements.txt [hunk 2]
```diff
@@ -127,11 +134,13 @@ opensearch-py==2.4.2
...
 packaging==22.0
-    # via
-    #   marshmallow
-    #   python-redis-rate-limit
+    # via python-redis-rate-limit
 pandas==2.2.1
     # via -r requirements.in
+platformdirs==4.2.0
+    # via virtualenv
+pre-commit==3.7.0
```
Comment (oliverjwroberts) [inline requirements.txt:142]: Is pre-commit needed in the normal/non-dev requirements?

**Review by cgsunkel — APPROVED** [review]

**Review by oliverjwroberts — APPROVED** [review]
Just a couple of comments, but looks good to me otherwise.

File: requirements.txt (hunk 2 above)
Comment (marijnkampf) [reply requirements.txt:142]: If I interpret the structure of the requirements.txt correctly pre-commit is required by `django-log-formatter-asim`

**Pull request merged**